h2/src/test/org/h2/test/unit/TestOldVersion.java

```diff
@@ -114,6 +113,10 @@ private void checkResult(ResultSet rs) throws SQLException {
     }
 
     private void testOldClientNewServer() throws Exception {
+        // TODO fails in pagestore mode
+        if (!config.mvStore) {
+            return;
```

Comment:
The whole test is PageStore-only. It should not be disabled, especially in such way.

I sent a possible fix in #1123.

Reply:
I merged a fix. `TestOldVersion` now completes normally in PageStore mode on Java 8 and Java 9. I didn't test Java 7, however.

So this test can be enabled again.